### PR TITLE
Initial release

### PR DESCRIPTION
**Pull request by twolfson**
This is the initial release of our `api-pagination` library. Unfortunately, initial releases are hard to break down since everything plays off each other (e.g. docs/lib/tests). In this PR:
- Added library
- Added tests
- Added documentation

/cc @brettlangdon

### CODE REVIEW

**Comment by twolfson** [comment]
I have been frustrated with how reStructuredText has been styling my nested lists. During research yesterday, I found out my current lists could be styled properly via a lot of whitespace:

``` rst
-    ret_val ``dict`` - Container for information about overall information and page specific information

    -    overall ``dict`` - Container for overall information

        - first_page ``int`` - Human based index for first page
```

The `-'s` need to line up with the next ones new line **and** there needs to be a newline between every new nesting.

After reflection today, I realized that the new line is necessary since reStructuredText supports things like blockquoting in a nested list so it needs to delimit. Additionally, we can use 2 spaces instead of 4 spaces for indentation for nesting which makes things more sane:

Before landing this PR, I will likely update the documentation to properly use `ul's`:

``` rst
- ret_val ``dict`` - Container for information about overall information and page specific information

  - overall ``dict`` - Container for overall information

    - first_page ``int`` - Human based index for first page
```

File: api_pagination/__init__.py [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+        """
+        Calculates the amount of pages that exist
+
+        Note: If your data has 0 items per page, you will receive a `ZeroDivisionError`
+
+        :rtype: int
+        :returns: Overall amount of pages (e.g. total=50, items_per_page=10 -> 5 pages)
+        """
+        # DEV: We must perform float division since we want to always return partial pages
+        #   (e.g. total=5 but items=1, we should still have 1 page
```
Comment (brettlangdon) [inline api_pagination/__init__.py]: this comment is backwards, we want `total=1` and `items=5` so the division will be `ceil(1/5)` to yield 1 page.

also missing the closing paren.

**Comment by brettlangdon** [comment]
comment about a comment, and yeah, the format of the lists looks a little wonky, if you update both of those then :+1:

File: api_pagination/__init__.py (hunk 1 above)
Comment (twolfson) [inline api_pagination/__init__.py]: Yea, I confused myself with `items` instead of `items_per_page`. Terminology is heard.

**Comment by twolfson** [comment]
Updated the comment and list formatting. It feels weird having a combination of 2 and 4 spaces but it feels weird having lists with no excess whitespace after bullets and some having them. I guess it's all tradeoffs and complaints about this not being markdown =/

**Comment by brettlangdon** [comment]
:+1:

**Pull request merged**